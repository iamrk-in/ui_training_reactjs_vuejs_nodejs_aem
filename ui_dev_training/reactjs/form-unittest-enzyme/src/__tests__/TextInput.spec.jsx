import React from "react";
import renderer from "react-test-renderer";

import { mount } from "enzyme";
import enzymeConfig from "../../enzymeConfig";
import TextInput from "../components/TextInput";


describe("TextInput", function() {
    
    it("should render correctly", function() {
        const tree = renderer.create(<TextInput />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should have div with correct class name', function() {
        const component = mount(<TextInput />);
        expect(component.find('div').hasClass('form-group')).toBe(true);
    });

    it('should have label with default title', function() {
        const component = mount(<TextInput />);
        expect(component.find('label').text().toEqual('text input'));
    });

    it('should call changeHandler onChange on input', function() {
        const mockHandler = jest.fn();
        const component = mount(<TextInput handleChange={mockHandler} />);
        const input = component.find('input');
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });

});