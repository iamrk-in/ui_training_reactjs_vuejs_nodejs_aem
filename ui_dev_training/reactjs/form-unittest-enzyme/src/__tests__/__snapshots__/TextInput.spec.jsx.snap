// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="text input"
  >
    Text Input
  </label>
  <input
    className="form-control"
    id="input"
    name="input"
    onChange={[Function]}
    type="input"
    value="hello world"
  />
</div>
`;
